### PR TITLE
* UPDATE add demo as default compiling option

### PR DESCRIPTION
**Pull request by JaylinYu**

### CODE REVIEW

**Review by wanghaEMQ — DISMISSED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -267,6 +267,7 @@ endif ()
 #  Build the tools
 
 add_subdirectory(docs/man)
+add_subdirectory(demo)
```
Comment (alvin1221) [inline CMakeLists.txt:270]: I suggest add an cmake option to decide build demos or not

**Review by alvin1221 — CHANGES_REQUESTED** [review]
cmake failed

**Review by aguasonic — APPROVED** [review]
Yes, building the demo code is useful!

**Comment by wanghaEMQ** [comment]
Did you overwrite your commits? The changes in CMakeLists.txt disappeared

**Comment by JaylinYu** [comment]
done by #225

**Pull request closed without merging**